3rd-party/class.jetpack-amp-support.php

```diff
@@ -78,6 +78,11 @@ function_exists( 'amp_is_canonical' ) // this is really just testing if the plug
 		return apply_filters( 'jetpack_is_amp_request', $is_amp_request );
 	}
 
+	static function has_amp_suffix() {
+		$request_path = parse_url( $_SERVER['REQUEST_URI'], PHP_URL_PATH );
+		return $request_path && substr_compare( $request_path, '/amp/', -4, 5, true );
```

Comment:
This wouldn't handle the case when a user has filtered `user_trailingslashit` to not use trailing slashes. I think it would be better to do:

```php
return preg_match( '#/amp/?$#i', $request_path );
```

This will ensure that the slash is optional.

Reply:
👍